File: fase-01/disciplina-02-docker/desafio-individual/spring-app/src/main/kotlin/com/olegon/fiap/soat/app/controller/ProgrammingLanguageController.kt
package com.olegon.fiap.soat.app.controller

import com.olegon.fiap.soat.app.repository.ProgrammingLanguangeRepository
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class ProgrammingLanguageController(
    private val programmingLanguangeRepository: ProgrammingLanguangeRepository
) {
    @GetMapping("/v1/programming-languages")
    fun listProgrammingLanguages() = programmingLanguangeRepository.findAll().map {
        it.Name
    }
}

File: fase-01/disciplina-02-docker/desafio-individual/dotnet-app/Controllers/ProgrammingLanguageController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using Olegon.Fiap.Soat.App.Entity;
using Olegon.Fiap.Soat.App.Model;
using Olegon.Fiap.Soat.App.Repository;

namespace Olegon.Fiap.Soat.App.Controllers;

[ApiController]

public class ProgrammingLanguageController : ControllerBase
{
    private readonly ProgrammingLanguageContext _dbContext;

    public ProgrammingLanguageController(ProgrammingLanguageContext dbContext)
    {
        this._dbContext = dbContext;
    }

    [HttpGet("/v1/programming-languages")]
    public Task<List<string>> ListProgrammingLanguages() => _dbContext.ProgrammingLanguages.Select(t => t.Name).ToListAsync();

    [HttpPost("/v1/programming-languages")]
    public async Task<ProgrammingLanguage> InsertProgrammingLanguage([FromBody] CreateProgrammingLanguageRequest payload)
    {
        var entry = _dbContext.Add<ProgrammingLanguage>(payload);

        await _dbContext.SaveChangesAsync();

        return entry.Entity;
    }
}
